File: 4deepseek_sol2.cpp
#include <iostream>
#include <vector>
#include <numeric>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<long long> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    long long sum = accumulate(a.begin(), a.end(), 0LL);
    cout << (sum % n == 0 ? "YES" : "NO") << "\n";
    return 0;
}

File: 4deepseek_sol2.java
import java.util.Scanner;
import java.util.stream.LongStream;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        long[] arr = new long[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextLong();
        }
        long sum = LongStream.of(arr).sum();
        System.out.println(sum % n == 0 ? "YES" : "NO");
        sc.close();
    }
}
